### ISSUE TITLE
Compile errors when defining nested buildable types in generic outer classes

### ISSUE THREAD

**Issue by tullisar** [google/FreeBuilder#268]
Defining a static buildable type in a generic outer class generates compiler errors when one of the fields is another nested class in the same file.

```java
public class GenericOuterBug<T> 
{
    @FreeBuilder
    public static abstract class Buildable
    {   
        public static class Builder extends GenericOuterBug_Buildable_Builder {}
        public abstract Sibling fieldWithError();
    }
    public static interface Sibling {}
}
```

The following is a portion of what gets generated. Most of the references to `GenericOuterBug.Sibling` get generated as `GenericOuterBug<T>.Sibling`:

```java
  private GenericOuterBug<T>.Sibling fieldWithError;
  private final EnumSet<GenericOuterBug_Buildable_Builder.Property> _unsetProperties =
      EnumSet.allOf(GenericOuterBug_Buildable_Builder.Property.class);
```

**Comment by chrisalice** [comment]
Argh. Generics. Thanks for the report, I'll try and get on this tomorrow :)

**Comment by chrisalice** [comment]
Hi @tullisar. I'm unable to reproduce this bug with the example code you've given. What version of FreeBuilder are you on? What version of Java, and what build system? Did you accidentally remove some part of your code that triggers the bug in condensing it for this report?

**Comment by tullisar** [comment]
Freebuilder Version: 1.14.6
JDK: 1.8.0_144
Using Eclipse Oxygen (4.7.1) to build. 

Looking at this a bit more, using Eclipse to build seems to trigger this. When I run a Maven build it seems to be just fine.

**Comment by tullisar** [comment]
I'm starting to suspect this might be a regression in Eclipse? I never recalled having this bug until recently (on the same construct I've listed as causing the bug). On a whim I launched Eclipse Mars (4.5.1) and ran a build and it generated the code just fine.

Since the maven build works I've just gone ahead and turn off Eclipse's automatic build for the time being. I'm not exactly sure what kind of bug report I'd even submit for Eclipse though. I'm not really familiar with annotation processors and how Eclipse interacts with them.

**Comment by chrisalice** [comment]
I can work around most things in FreeBuilder's code, so let's see if I can repro now I know the IDE version. Thanks :)

**Comment by chrisalice** [comment]
Yep, I can reproduce this now. (Irritatingly I _can't_ reproduce it in a test, but I'll figure _something_ out to try and avoid future regressions.)

**Comment by chrisalice** [comment]
Sorry, @tullisar, I haven't got around to prioritising this yet. Hard to motivate myself to work around a niche Eclipse bug :(
